flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/SlotSharingManagerTest.java

```diff
@@ -0,0 +1,502 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.jobmaster.slotpool;
+
+import org.apache.flink.runtime.clusterframework.types.AllocationID;
+import org.apache.flink.runtime.executiongraph.utils.SimpleAckingTaskManagerGateway;
+import org.apache.flink.runtime.instance.SimpleSlotContext;
+import org.apache.flink.runtime.instance.SlotSharingGroupId;
+import org.apache.flink.runtime.jobmanager.scheduler.Locality;
+import org.apache.flink.runtime.jobmanager.slots.DummySlotOwner;
+import org.apache.flink.runtime.jobmaster.LogicalSlot;
+import org.apache.flink.runtime.jobmaster.SlotContext;
+import org.apache.flink.runtime.jobmaster.SlotRequestId;
+import org.apache.flink.runtime.taskmanager.LocalTaskManagerLocation;
+import org.apache.flink.util.AbstractID;
+import org.apache.flink.util.FlinkException;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.Test;
+
+import java.util.Collections;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Test cases for the {@link SlotSharingManager}.
+ */
+public class SlotSharingManagerTest extends TestLogger {
+
+	private static final SlotSharingGroupId slotSharingGroupId = new SlotSharingGroupId();
+
+	private static final DummySlotOwner slotOwner = new DummySlotOwner();
+
+	private static final TestingAllocatedSlotActions allocatedSlotActions = new TestingAllocatedSlotActions();
+
+	@Test
+	public void testRootSlotCreation() {
+		final SlotSharingManager slotSharingManager = new SlotSharingManager(
+			slotSharingGroupId,
+			allocatedSlotActions,
+			slotOwner);
+
+		SlotRequestId slotRequestId = new SlotRequestId();
+		SlotRequestId allocatedSlotRequestId = new SlotRequestId();
+
+		final SlotSharingManager.MultiTaskSlot multiTaskSlot = slotSharingManager.createRootSlot(
+			slotRequestId,
+			new CompletableFuture<>(),
+			allocatedSlotRequestId);
+
+		assertEquals(slotRequestId, multiTaskSlot.getSlotRequestId());
+		assertNotNull(slotSharingManager.getTaskSlot(slotRequestId));
+	}
+
+	@Test
+	public void testRootSlotRelease() throws ExecutionException, InterruptedException {
+		final CompletableFuture<SlotRequestId> slotReleasedFuture = new CompletableFuture<>();
+		final TestingAllocatedSlotActions testingAllocatedSlotActions = new TestingAllocatedSlotActions();
```

Comment:
You already have an instance of that defined in line `57`.

Reply:
Indeed. Will change it.